### PR TITLE
Implement `layer.ready` and `movie.ready` events

### LINKED ISSUE #147: Add 'layer.ready' event for layers that implement `ready`
This will complement the new `ready` property. It should fire every time `ready` changes from false to true.

The following layers and layer mixins implement `ready`:
- `AudioSourceMixin`
- `VisualSourceMixin`
- `Visual`

### PR DESCRIPTION
**Pull request by zbendas**
Should resolve  #147 and #148

### CODE REVIEW

**Comment by clabe45** [comment]
Thanks for contributing!

Initially, I was thinking the events should be triggered by other events, like so:
- `movie.ready` is emitted when either `layer.ready` or `effect.ready` are emitted if all its layers and effects are ready
- For `etro.layer.Visual`, `layer.ready` is emitted when all its effects are ready (similar to movies)
- For `etro.layer.VisualSource`, `layer.ready` is emitted when its source emits an `loadeddata` DOM event and its effects are ready (something along those lines)

The advantage of doing this over the way you implemented in the PR is that if the `ready` getter isn't accessed one frame, the event will still be published.

In hindsight, I see that my proposed implementation wouldn't work very well, because I don't see a way to make this work with class inheritance. For `VisualSource` to emit the event when `loadeddata` fires *and also* when its effects are ready, it would have to re-implement the logic to listen to effects. There's no way to listen to Etro events published from a superclass without listening to events published from its own class. This would result in a lot of repeated code, which doesn't sound like a great idea.

I still don't think it's a good idea to put the event publishing logic in the `ready` getters, since this could result in buggy behavior in the future. Perhaps it isn't worth it to add these events at all. The event system is not something I'm that fond of, and I've been meaning to deprecate a lot of events. Maybe it would make sense to add this as an async method, not an event. We could add a public movie method `waitUntilReady()` that returns a promise that resolves when the current frame is loaded (`movie.ready` is true). This method could check if the movie is ready every, say, animation frame.

In short, I think we should add a public async method to the movie that waits until it is ready, instead of adding the events. I'm open to input if you have thoughts on any of this.

**Comment by clabe45** [comment]
Thanks again for working on this, but I closed the ready event issues today, because I don't think it's possible with the event system.
